AuthServ/AuthServer_Private.h

```diff
@@ -105,23 +106,20 @@ struct AuthServer_Private : public AuthClient_Private
     uint32_t m_acctFlags;
     AuthServer_PlayerInfo m_player;
     uint32_t m_ageNodeId;
-    std::map<uint32_t, DS::Stream*> m_downloads;
+    std::map<uint32_t, std::unique_ptr<DS::Stream>> m_downloads;
 
     AuthServer_Private() : m_serverChallenge(0), m_acctFlags(0), m_ageNodeId(0) { }
 
     ~AuthServer_Private()
     {
-        while (!m_downloads.empty()) {
-            auto item = m_downloads.begin();
-            delete item->second;
-            m_downloads.erase(item);
-        }
+        m_downloads.clear();
```

Comment:
This should happen automatically in the `std::map` destructor, so this destructor can be fully default now.

Reply:
I wasn't sure if we wanted to retain explicit clear on destruct or not since the previous constructor did explicitly remove each element from the map after deleting the stream. Will fix.